.tekton/tasks/verify-smoke-e2e.yaml

```diff
@@ -0,0 +1,88 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: stage-verify-e2e-test
+spec:
+  description: >-
+    This task is for running stage e2e test on RHTAP staging cluster
+  params:
+  - name: PLATFORM
+    description: which environment the test is running on.
+    default: "stage"
+  - name: TASK_RETRIES
+    description: number of times the task will be rerun
+    default: "0"
+  workspaces:
+    - name: stage-e2e-git
+  steps:
+    - name: run-stage-verify-e2e-test
+      image: registry.access.redhat.com/ubi8/go-toolset:1.19.10-10.1692783630
+      workingDir: $(workspaces.stage-e2e-git.path)
+      script: |
+        #!/usr/bin/env bash
+        set -o errexit
+        set -o nounset
+        set -o pipefail
+
+        mkdir -p ~/.kube
+        cat <<EOF > ~/.kube/config
+        apiVersion: v1
+        clusters:
+        - cluster:
+            server: YOUR_KUBE_API_SERVER_URL
+          name: YOUR_CLUSTER_NAME
+        contexts:
+        - context:
+            cluster: YOUR_CLUSTER_NAME
+            user: YOUR_USER_NAME
+          name: YOUR_CONTEXT_NAME
+        current-context: YOUR_CONTEXT_NAME
+        kind: Config
+        preferences: {}
+        users:
+        - name: YOUR_USER_NAME
+          user:
+            token: YOUR_KUBE_API_TOKEN
+        EOF
+        make build
+        E2E_ARGS_EXEC="--ginkgo.label-filter $(params.PLATFORM) --ginkgo.vv" make run
```

Comment:
I'm a little late in realizing this, as I have not had the bandwidth/expertise to fully examine @prietyc123 's work here, but one thing I just realized as part of trying to catch up a little, where I do know a bit about rhtap's e2e's, is that if https://github.com/sawood14012/e2e-tests/blob/verify-stage/tests/stage/verifystage.go is still the latest version of the smoke test, it is not covering any of the pipelines as code related e2e's like what @psturc (IIRC) create with the rhtap-demo suite. i.e. https://github.com/redhat-appstudio/e2e-tests/tree/main/tests/rhtap-demo

So
- perhaps this was already discussed amongst QE and I missed the outcome @prietyc123 .... was there discussion around using something like rhtap-demo for the smoke test, and it was dismissed?  or was it never considered?
- or is there a more recent version of smoke tests vs. what is in that `sawood14012` fork of e2e tests?
- @Roming22 how comfortable are you with gating tests for pipeline service *NOT* exercising the pipelines as code path, given the relative error rate of that component vs. other components from openshift pipelines?

Reply:
This PR adds extra validation on top of what is already there (i.e. tests in development, which do test PaC). It is true that it might not test PaC now,  but I would think that it could be added in the future.

IMO this PR moves things in the right direction even if it does not cover 100% of the usecases we would ultimately want to cover.

Reply:
@gabemontero I agree with @Roming22 . The flow Romain described is the plan for the promotion test. To see the plan on test related story please find the details [here](https://issues.redhat.com/browse/RHTAP-1743)

Reply:
thanks for the updates and filling in some of my blanks wrt things here @prietyc123 @Roming22 sounds like we are good here